custom-completions/rustup/rustup.nu

```diff
@@ -0,0 +1,201 @@
+ 
+def "nu-complete rustup" [] {
+  ^rustup --help 
+  | str replace --regex --multiline '(rustup[\s\S]*(?=SUBCOMMANDS:))' '' 
+  | str replace --regex --multiline '\n+DISCUSSION[\s\S]*' ''
+  | lines 
+  | where $it starts-with "   " 
+  | parse -r '\s*(?P<value>[^ ]+) \s*(?P<description>\w.*)'
+}
+
+def "nu-complete rustup toolchain" [] {
+  ^rustup toolchain 
+  | str replace --regex --multiline '(rustup[\s\S]*(?=SUBCOMMANDS:))' '' 
+  | str replace --regex --multiline '\n+DISCUSSION[\s\S]*' ''
+  | lines 
+  | where $it starts-with "   " 
+  | parse -r '\s*(?P<value>[^ ]+) \s*(?P<description>\w.*)'
+}
+
+
+def "nu-complete rustup toolchain list" [] {
+  ^rustup toolchain list
+  | lines
+  # TODO: add `stable` `nightly` and any other starndard toolchain
+}
+
+def "nu-complete rustup target" [] {
+  ^rustup target 
+  | str replace --regex --multiline '(rustup[\s\S]*(?=SUBCOMMANDS:))' '' 
+  | str replace --regex --multiline '\n+DISCUSSION[\s\S]*' ''
+  | lines 
+  | where $it starts-with "   " 
+  | parse -r '\s*(?P<value>[^ ]+) \s*(?P<description>\w.*)'
+}
+
+def "nu-complete rustup target list" [] {
+  ^rustup target list
+  | lines
+}
+
+def "nu-complete rustup target list --installed" [] {
+  ^rustup target list --installed
+  | lines
+}
+
+def "nu-complete rustup update" [] {
+  ^rustup toolchain list
+  | lines
+}
+
+def "nu-complete rustup component" [] {
+  ^rustup component 
+  | str replace --regex --multiline '(rustup[\s\S]*(?=SUBCOMMANDS:))' '' 
+  | str replace --regex --multiline '\n+DISCUSSION[\s\S]*' ''
+  | lines 
+  | where $it starts-with "   " 
+  | parse -r '\s*(?P<value>[^ ]+) \s*(?P<description>\w.*)'
+}
+
+def "nu-complete rustup component list" [] {
+  ^rustup component list
+  | lines
+}
+
+def "nu-complete rustup component list installed" [] {
+  ^rustup component list 
+  | lines 
+  | find --regex '^.+(?=\s\(installed)'
+}
+
+def "nu-complete rustup override" [] {
+  ^rustup override 
+  | str replace --regex --multiline '(rustup[\s\S]*(?=SUBCOMMANDS:))' '' 
+  | str replace --regex --multiline '\n+DISCUSSION[\s\S]*' ''
+  | lines 
+  | where $it starts-with "   " 
+  | parse -r '\s*(?P<value>[^ ]+) \s*(?P<description>\w.*)'
+}
+
+def "nu-complete rustup override list" [] {
+  ^rustup override list
+  | lines
+  # TODO: I don't have any override to match
+  # TODO: exclude the result "no overrides"
+}
+
+def "nu-complete rustup override list installed" [] {
+  ^rustup override list 
+  | lines 
+  # TODO: I don't have any override to match
+  # TODO: exclude the result "no overrides"
+}
+
+def "nu-complete rustup self" [] {
+  ^rustup self 
+  | str replace --regex --multiline '(rustup[\s\S]*(?=SUBCOMMANDS:))' '' 
+  | str replace --regex --multiline '\n+DISCUSSION[\s\S]*' ''
+  | lines 
+  | where $it starts-with "   " 
+  | parse -r '\s*(?P<value>[^ ]+) \s*(?P<description>\w.*)'
+}
+
+# ------------------ export extern commands ------------------
+
+export extern "rustup" [
```

Comment:
@AucaCoyan, great to see more completion. I wanted to try that one by when I `use` it,  I get following error:

![grafik](https://github.com/nushell/nu_scripts/assets/16558417/76ad3af1-e795-4d97-8987-6501daabdd73)

Is that something that works with the latest development version of nu?

Reply:
Mmmm I don't know how to fix that error. In my `config.nu` I use `source` instead of `use`
```nu
source ~/other-repos/nu/nu_scripts/custom-completions/rustup/rustup.nu
```
can you try that?

Reply:
Oh, sorry! You intended to fix this error with the `export extern "rustup" [...]`. Now I got it 😋

Reply:
I get the same error on the latest main, which I believe means this module isn't constructed correctly. @amtoine any tips on how to fix this problem?

Reply:
`source ../../rustup.nu` works for me. @fdncred, is that a bug in Nushell? What should be prefered for loading completions? `use` or `source`?

Reply:
I don't think so. I think modules are not supposed to be named the same name as the file.

Reply:
Yes, I know what you mean. Python has it too! 
This is obviously the product of me trying to write my first completions script. But honestly I am surprised the level of depth I had to achieve to stumble into a design issue like this. I wasn't expecting to talk about this!
pat on the back for me 😄.

Give me a couple of hours/days and I will fix it 🔧

Reply:
I did a PR #696 here